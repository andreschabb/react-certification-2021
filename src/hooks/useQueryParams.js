import { useEffect, useState } from 'react';
import { useLocation } from 'react-router-dom';

const useQueryParams = () => {
  const [searchedText, setSearchedText] = useState('');
  const [videoId, setVideoId] = useState('');
  const location = useLocation();

  useEffect(() => {
    const query = new URLSearchParams(location.search);
    setSearchedText(query.get('searchedText'));
    setVideoId(query.get('id'));
  }, [location]);

  return { searchedText, videoId };
};

export default useQueryParams;
