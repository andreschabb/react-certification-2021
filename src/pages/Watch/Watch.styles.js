import styled from 'styled-components';

export const Container = styled.section`
  height: 95vh;
  width: 98%;
`;
