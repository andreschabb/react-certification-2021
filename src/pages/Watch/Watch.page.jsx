import React from 'react';
import { useLocation } from 'react-router-dom';

import { Container } from './Watch.styles';
import VideoDetails from '../../components/VideoDetails';
import useQueryParams from '../../hooks/useQueryParams';

const WatchPage = () => {
  const queryParams = useQueryParams();
  const location = useLocation();

  const relatedVideos =
    location.state && location.state.videos ? location.state.videos : [];
  const descriptionVideo =
    location.state && location.state.description ? location.state.description : '';
  const titleVideo = location.state && location.state.title ? location.state.title : '';

  return (
    <Container>
      <VideoDetails
        videoId={queryParams.videoId}
        title={titleVideo}
        description={descriptionVideo}
        videos={relatedVideos}
      />
    </Container>
  );
};

export default WatchPage;
