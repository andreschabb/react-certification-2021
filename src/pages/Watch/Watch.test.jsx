import React from 'react';
import { useHistory, useLocation } from 'react-router';
import { render, screen } from '@testing-library/react';

import useQueryParams from '../../hooks/useQueryParams';
import Watch from './Watch.page';

const mockedVideos = [
  {
    id: { kind: 'youtube#video', videoId: 'nmXMgqjQzls' },
    snippet: {
      title: 'Video Tour | Welcome to Wizeline Guadalajara',
      description:
        'Follow Hector Padilla, Wizeline Director of Engineering, for a lively tour of our office. In 2018, Wizeline opened its stunning new office in Guadalajara, Jalisco, ...',
      thumbnails: {
        default: {
          url: 'https://i.ytimg.com/vi/nmXMgqjQzls/default.jpg',
          width: 120,
          height: 90,
        },
      },
    },
  },
  {
    id: { kind: 'youtube#video', videoId: 'nmXMgqjQlhj' },
    snippet: {
      title: 'Title',
      description: 'Description',
      thumbnails: {
        default: {
          url: 'https://i.ytimg.com/vi/nmXMgqjQzls/default.jpg',
          width: 120,
          height: 90,
        },
      },
    },
  },
];

jest.mock('../../hooks/useQueryParams');
jest.mock('react-router');

describe('Watch', () => {
  it('should renders', () => {
    useQueryParams.mockImplementation(() => ({ videoId: 'VideoId' }));
    useLocation.mockImplementation(() => ({
      state: { title: 'Title', description: 'Description', videos: mockedVideos },
    }));
    useHistory.mockImplementation(() => ({
      push: jest.fn(),
    }));
    const { container } = render(<Watch />);

    expect(container).toMatchSnapshot();
  });

  it('should renders without state', () => {
    useQueryParams.mockImplementation(() => ({ videoId: 'VideoId' }));
    useLocation.mockImplementation(() => ({ state: {} }));
    useHistory.mockImplementation(() => ({
      push: jest.fn(),
    }));

    render(<Watch />);

    expect(screen.getByText('Add To Favorites')).toBeTruthy();
  });
});
