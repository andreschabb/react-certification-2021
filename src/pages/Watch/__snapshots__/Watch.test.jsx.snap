// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Watch should renders 1`] = `
<div>
  <section
    class="sc-bdfBwQ gTLTiV"
  >
    <div
      class="sc-eCssSg gIaHdG"
    >
      <div
        class="sc-jSgupP eAqXwZ"
      >
        <iframe
          allowfullscreen=""
          class="sc-iBPRYJ kFCpgj"
          src="https://www.youtube.com/embed/VideoId"
          title="video-player"
        />
        <div
          class="sc-fubCfw nHWfQ"
        >
          <div
            class="sc-pFZIQ ldojFV"
          >
            Title
          </div>
          <button
            class="sc-jrAGrp dIMBWR"
            data-testid="favorites-button"
          >
            Add To Favorites
            <i
              aria-hidden="true"
              class="favorite icon"
            />
          </button>
        </div>
        <hr />
        <p
          class="sc-kEjbxe frTmbO"
        >
          Description
        </p>
      </div>
      <div
        class="sc-gKsewC dcfKuV"
      >
        <div
          class="sc-gsTCUz dcMcIP"
          data-testid="related-video-item"
        >
          <img
            alt="Video Tour | Welcome to Wizeline Guadalajara"
            class="sc-dlfnbm gXZbgX"
            src="https://i.ytimg.com/vi/nmXMgqjQzls/default.jpg"
          />
          <p
            class="sc-hKgILt cXOrwW"
          >
            Video Tour | Welcome to Wizeline Guadalajara
          </p>
        </div>
        <div
          class="sc-gsTCUz dcMcIP"
          data-testid="related-video-item"
        >
          <img
            alt="Title"
            class="sc-dlfnbm gXZbgX"
            src="https://i.ytimg.com/vi/nmXMgqjQzls/default.jpg"
          />
          <p
            class="sc-hKgILt cXOrwW"
          >
            Title
          </p>
        </div>
      </div>
    </div>
  </section>
</div>
`;
